common/testData/personNameTest/de.txt

```diff
@@ -0,0 +1,728 @@
+# CLDR person name formatting test data for: de
+#
+# Test lines have the following structure:
```

Comment:
Suggestion: It would be nice if for new testdata files, if CLDR and ICU would start using a more structured syntax. TOML is a good one. Here's one way that the first set of cases could look like in TOML:

```toml
[[case]]
given = "Lena"

[[case.expectation]]
result = "Lena"
parameters = [
    { ord = "givenFirst", len = "long", usage = "addressing", form  = "formal" },
    { ord = "givenFirst", len = "long", usage = "addressing", form  = "informal" },
    { ord = "givenFirst", len = "long", usage = "referring", form  = "formal" },
    { ord = "givenFirst", len = "long", usage = "referring", form  = "informal" },
    { ord = "givenFirst", len = "medium", usage = "addressing", form  = "formal" },
    { ord = "givenFirst", len = "medium", usage = "addressing", form  = "informal" },
    { ord = "givenFirst", len = "medium", usage = "referring", form  = "formal" },
    { ord = "givenFirst", len = "medium", usage = "referring", form  = "informal" },
    { ord = "givenFirst", len = "short", usage = "addressing", form  = "formal" },
    { ord = "givenFirst", len = "short", usage = "addressing", form  = "informal" },
    { ord = "givenFirst", len = "short", usage = "referring", form  = "formal" },
    { ord = "givenFirst", len = "short", usage = "referring", form  = "informal" },
    { ord = "sorting", len = "long", usage = "referring", form  = "formal" },
    { ord = "sorting", len = "long", usage = "referring", form  = "informal" },
    { ord = "sorting", len = "medium", usage = "referring", form  = "formal" },
    { ord = "sorting", len = "medium", usage = "referring", form  = "informal" },
    { ord = "sorting", len = "short", usage = "referring", form  = "formal" },
    { ord = "sorting", len = "short", usage = "referring", form  = "informal" },
    { ord = "surnameFirst", len = "long", usage = "addressing", form  = "formal" },
    { ord = "surnameFirst", len = "long", usage = "addressing", form  = "informal" },
    { ord = "surnameFirst", len = "long", usage = "referring", form  = "formal" },
    { ord = "surnameFirst", len = "long", usage = "referring", form  = "informal" },
    { ord = "surnameFirst", len = "medium", usage = "addressing", form  = "formal" },
    { ord = "surnameFirst", len = "medium", usage = "addressing", form  = "informal" },
    { ord = "surnameFirst", len = "medium", usage = "referring", form  = "formal" },
    { ord = "surnameFirst", len = "medium", usage = "referring", form  = "informal" },
    { ord = "surnameFirst", len = "short", usage = "addressing", form  = "formal" },
    { ord = "surnameFirst", len = "short", usage = "addressing", form  = "informal" },
    { ord = "surnameFirst", len = "short", usage = "referring", form  = "formal" },
    { ord = "surnameFirst", len = "short", usage = "referring", form  = "informal" },
]

[[case.expectation]]
result = "L"
parameters = [
    { ord = "givenFirst", len = "long", usage = "monogram", form = "formal" },
    { ord = "givenFirst", len = "long", usage = "monogram", form = "informal" },
    { ord = "givenFirst", len = "medium", usage = "monogram", form = "formal" },
    { ord = "givenFirst", len = "medium", usage = "monogram", form = "informal" },
    { ord = "givenFirst", len = "short", usage = "monogram", form = "formal" },
    { ord = "givenFirst", len = "short", usage = "monogram", form = "informal" },
    { ord = "surnameFirst", len = "long", usage = "monogram", form = "formal" },
    { ord = "surnameFirst", len = "long", usage = "monogram", form = "informal" },
    { ord = "surnameFirst", len = "medium", usage = "monogram", form = "formal" },
    { ord = "surnameFirst", len = "medium", usage = "monogram", form = "informal" },
    { ord = "surnameFirst", len = "short", usage = "monogram", form = "formal" },
    { ord = "surnameFirst", len = "short", usage = "monogram", form = "informal" },
]
```

Reply:
If you don't like repeating the keys, it's _fine_ to just make them a string list, like

```toml
[[case.expectation]]
result = "L"
parameters = [
    [ "givenFirst", "long", "monogram", "formal" ],
    [ "givenFirst", "long", "monogram", "informal" ],
    [ "givenFirst", "medium", "monogram", "formal" ],
    [ "givenFirst", "medium", "monogram", "informal" ],
    [ "givenFirst", "short", "monogram", "formal" ],
    [ "givenFirst", "short", "monogram", "informal" ],
    [ "surnameFirst", "long", "monogram", "formal" ],
    [ "surnameFirst", "long", "monogram", "informal" ],
    [ "surnameFirst", "medium", "monogram", "formal" ],
    [ "surnameFirst", "medium", "monogram", "informal" ],
    [ "surnameFirst", "short", "monogram", "formal" ],
    [ "surnameFirst", "short", "monogram", "informal" ],
]
```

Reply:
I thought of having more structured data. But for anyone that doesn't have access to that structured data processor, it is actually harder to parse than just a simple file. (And it is more voluminous, not that that is a big factor.)

The main advantage I can see to the structure is that is a bit more robust in the face of changes.

Reply:
Acknowledged

In languages that have easy access to libraries to help with structured data parsing, like Rust and JavaScript, it results in better hygiene than having to write a custom plain file parser by hand. So we're improving the experience in C++ at the cost of reduced hygiene in Rust.

TOML is nice because I feel it is a good middle ground. A custom parser can be written for it, a bit more tricky than the plain-txt-file approach, but still totally doable. But, for languages that have upgraded support for structured data types, we get the fully hygienic approach.

Reply:
For my part, this format looks fine.  It's a little verbose, but should be easy to parse.  I can work with it.

Reply:
Please I'm really interested in this program, please someone should help 🙏